### PR TITLE
add regression demo config ,fixed #10

### LINKED ISSUE #10: example configuration for regression.
- In this example, we would like to show how the do regression task in PaddlePaddle.

- Please refer to [this configuration file](https://github.com/lcy-seso/paddle_confs_v1/blob/master/regression/regression.conf) written in old PaddlePaddle API.

- Note that, currently PaddlePaddle provides two regression costs: mse and [huber loss](http://www.paddlepaddle.org/develop/doc/api/v2/config/layer.html#huber-cost). Please consider adding an example for huber loss.

- Please pull your codes and docs into the [regressoin](https://github.com/PaddlePaddle/models/tree/develop/regression) directory.

### PR DESCRIPTION
**Pull request by chrisxu2016**

### CODE REVIEW

File: regression/regression.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    #input2 architecture
+    input2 = paddle.layer.data(
+        name='input2_word',
+        type=paddle.data_type.integer_value_sequence(input2_dict_dim))
+
+    with paddle.layer.mixed(size=word_emb_dim, bias_attr=False) as input2_emb:
+        input2_emb += paddle.layer.table_projection(
+            input=input2,
+            param_attr=paddle.attr.Param(name='_emb_basic', is_static=True))
```
Comment (lcy-seso) [inline regression/regression.py]: line 16 to 19, please use `paddle.layer.embedding`.

File: regression/regression.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    input2_vec = paddle.layer.pooling(
+        input=input2_emb,
+        pooling_type=paddle.pooling.Sum(),
+        #act=paddle.activation.Tanh(),
```
Comment (lcy-seso) [inline regression/regression.py]: Please remove the comments if it is useless.

File: regression/regression.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        bias_attr=paddle.attr.ParameterAttribute(
+            name='_avg.bias_basic', is_static=True))
+
+    hidden_input2 = paddle.layer.fc(
+        input=input2_vec,
+        size=hidden_dim,
+        act=paddle.activation.Tanh(),
+        param_attr=paddle.attr.Param(name='_hidden_input2.w', is_static=True),
+        bias_attr=paddle.attr.ParameterAttribute(
+            name='_hidden_input2.bias', is_static=True))
```
Comment (lcy-seso) [inline regression/regression.py]: please make line 5 ~ 34 a function and make `is_static` a parameter, because it is repeated with line 37 ~ 59.

File: regression/regression.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+def main():
+    paddle.init(use_gpu=False, trainer_count=1)
+    is_generating = False
+
+    #input1 and input2 dict dim
+    dict_size = 30000
+    input1_dict_dim = input2_dict_dim = dict_size
+
+    #train the network
+    if not is_generating:
```
Comment (lcy-seso) [inline regression/regression.py]: This example does not need to generate anything. `is_generating` is not appropriate.

File: regression/regression.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+        # define optimize method and trainer
+        optimizer = paddle.optimizer.Adam(
+            learning_rate=5e-5,
+            regularization=paddle.optimizer.L2Regularization(rate=8e-4))
+
+        trainer = paddle.trainer.SGD(
+            cost=cost, parameters=parameters, update_equation=optimizer)
+
+        # define data reader
```
Comment (lcy-seso) [inline regression/regression.py]: If to test whether the configuration can run, it is no problem to use the WMT14 machine translation dataset, but it is inappropriate to be used in the examples to the users, because such inputs may not reasonable. I suggest giving several example data.

**Review by lcy-seso — CHANGES_REQUESTED** [review]
request changes.

File: regression/regression.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
+import paddle.v2 as paddle
+import sys
+
+
+def regression_net(input1_dict_dim, input2_dict_dim, is_generating=False):
```
Comment (lcy-seso) [inline regression/regression.py]: `is_generating` is not used in the configuration, so it can be removed.

File: regression/regression.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    ### Network Architecture
+    word_vector_dim = 512
```
Comment (lcy-seso) [inline regression/regression.py]: `word_vector_dim` is not used in the configuration, so it can be removed.

File: regression/regression.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+    #input1 architecture
+    input1 = paddle.layer.data(
+        name='input1_word',
+        type=paddle.data_type.integer_value_sequence(input1_dict_dim))
+
+    with paddle.layer.mixed(size=word_emb_dim, bias_attr=False) as input1_emb:
+        input1_emb += paddle.layer.table_projection(
+            input=input1,
+            param_attr=paddle.attr.Param(name='emb_input1', initial_std=0.02))
```
Comment (lcy-seso) [inline regression/regression.py]: line 36 ~ 44, please use `paddle.layer.embedding`.

File: regression/regression.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+    input1_vec = paddle.layer.pooling(
+        input=input1_emb,
+        pooling_type=paddle.pooling.Sum(),
+        #act=paddle.activation.Tanh(),
```
Comment (lcy-seso) [inline regression/regression.py]: please remove the useless comment.

File: regression/regression.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    hidden_input1 = paddle.layer.fc(
+        input=input1_vec,
+        size=hidden_dim,
+        act=paddle.activation.Tanh(),
+        param_attr=paddle.attr.ParameterAttribute(
+            name='_hidden_input1.w', initial_std=0.03),
+        bias_attr=paddle.attr.ParameterAttribute(name='_hidden_input1.bias'))
+
+    cost = paddle.layer.mse_cost(input=hidden_input1, label=hidden_input2)
+    #cost = paddle.layer.huber_cost(input=hidden_input1, label=hidden_input2)
```
Comment (lcy-seso) [inline regression/regression.py]: Remove line 62. Different costs should be chosen by user defined parameter, or create an other function.

**Review by lcy-seso — CHANGES_REQUESTED** [review]
request changes.

**Comment by luotao1** [comment]
为了保证版本库相对清晰，请合并下commit次数，update readme.md更新太多次了。

File: regression/regression.py (hunk 1 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 2 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 3 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 4 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 5 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 6 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 7 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 8 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 9 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/regression.py (hunk 10 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

**Comment by chrisxu2016** [comment]
增加新的示例数据

File: regression/README.md [hunk 11]
```diff
@@ -1 +1,262 @@
-TBD
+
```
Comment (lcy-seso) [inline regression/README.md]: 第一行多余的空格去掉。

File: regression/README.md [hunk 12]
```diff
@@ -1 +1,262 @@
...
+# 回归问题
+
+回归问题是机器学习中的一个经典问题，主要目的是构建一个函数将输入数据与输出数据关联起来。本示例中拟合两条语意相近的语句，通过构建了一个相同结构的网络，对源数据与目标数据进行编码。迭代更新源数据网络的参数，来拟合目标数据的编码，完成了一个简单的回归问题。用户可以利用机器翻译中的[WMT-14](https://github.com/PaddlePaddle/book/tree/develop/08.machine_translation#数据介绍)数据集来测试该配置文件，经典的线性回归，请参考[fit_a_line](https://github.com/PaddlePaddle/book/tree/develop/01.fit_a_line).
```
Comment (lcy-seso) [inline regression/README.md]: 1. ”回归问题是机器学习中的一个经典问题，主要目的是构建一个函数将输入数据与输出数据关联起来“ 这句话这样下定义不准确，请修改一下。
2. 本示例中拟合两条语意相近的语句，通过构建了一个相同结构的网络，对源数据与目标数据进行编码。 -->  这个解释不对，和回归问题什么关系？请在这里介绍一下 static paramerter 的概念？
3. 用户可以利用机器翻译中的[WMT-14](https://github.com/PaddlePaddle/book/tree/develop/08.machine_translation#数据介绍)数据集来测试该配置文件， --> 请不要提机器翻译，原因之前解释过，这不是一个符合逻辑的例子。

4. 请扩充背景介绍一节，至少包括以下内容：
    1.  回归相关的一些背景和概念介绍。
    2. 回归问题的应用场景。
    3. 重新解释一下本节的例子。

File: regression/README.md [hunk 13]
```diff
@@ -1 +1,262 @@
...
+
+## 数据准备
+
+本示例模拟的是两条语句相近的两条语句的编码相似，具体示例如下图所示，左右两端语句的语意相近，中间以 `\t` 分割。
```
Comment (lcy-seso) [inline regression/README.md]: 1. 背景一节介绍之后，这里不需要再解释应用背景，直接介绍数据的格式即可~

File: regression/regression.py [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+import paddle.v2 as paddle
+import data_process
+import sys
+
+
+def encode_net(input_dict_dim, word_emb_dim, hidden_dim, is_static=False):
+    """
+        the input data is encoded to obtain the same size encoded vector
+        params:
```
Comment (lcy-seso) [inline regression/regression.py]: 注释的风格请和Paddle现有注释风格保持一致~ 可以参考这里。
 https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/data_sources.py#L35

File: regression/data_process.py [hunk 15]
```diff
@@ -0,0 +1,75 @@
+import tarfile
```
Comment (lcy-seso) [inline regression/data_process.py:1]: 配置目前有比较大的问题~ 我们先仔细沟通一下~

**Review by lcy-seso — CHANGES_REQUESTED** [review]
配置目前有比较大的问题。

File: regression/README.md (hunk 11 above)
Comment (chrisxu2016) [reply regression/README.md]: done

File: regression/README.md (hunk 12 above)
Comment (chrisxu2016) [reply regression/README.md]: done

File: regression/README.md (hunk 13 above)
Comment (chrisxu2016) [reply regression/README.md]: done

File: regression/regression.py (hunk 14 above)
Comment (chrisxu2016) [reply regression/regression.py]: done

File: regression/data_process.py (hunk 15 above)
Comment (chrisxu2016) [reply regression/data_process.py:1]: done，目前还差一个小的样例数据
